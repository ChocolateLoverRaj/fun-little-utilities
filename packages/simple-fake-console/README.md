# Simple Fake Console

Create fake consoles for testing purpose

## Usage

### Use it directly

```javascript
const { ConsoleInstance, ActionType, getString } = require('simple-fake-console')
const console = new ConsoleInstance()

console.log('hello', 'world')
console.info(0, 1, 2, 3)
console.error(new Error('foo'))
console.warn('Warning')

// Get all actions
//
// Result: Array<{
//   type: 'log' | 'info' | 'error' | 'warn',
//   data: any[]
// }>
console.getActions()

// Get string from selected types of actions
//
// Result: string
getString({
  console,
  types: [ActionType.Log, ActionType.Info]
})
```

### Use it in test via dependency injection

```typescript
// main.ts

import { Console } from 'simple-fake-console'

export function main (console: Console) {
  console.log('hello', 'world')
}
```

```typescript
// test.ts

import assert from 'assert'
import { ConsoleInstance, ActionType, getString } from 'simple-fake-console'
import { main } from './main'

const fakeConsole = new ConsoleInstance()
main(fakeConsole)

assert.deepStrictEqual(
  fakeConsole.getActions(),
  [{
    type: 'log',
    data: ['hello', 'world']
  }]
)

assert.strictEqual(
  getString({ console, types: [ActionType.Log] }),
  'hello world'
)
```

## License

[MIT](https://git.io/fxKXN) © [Hoàng Văn Khải](https://github.com/KSXGitHub)
