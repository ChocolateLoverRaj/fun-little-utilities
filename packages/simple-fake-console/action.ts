export enum ActionType {
  Log = 'log',
  Info = 'info',
  Error = 'error',
  Warn = 'warn'
}

export type ActionData = ReadonlyArray<any>

export interface Action {
  type: ActionType
  data: ActionData
}

export class ActionInstance implements Action {
  constructor (
    public type: ActionType,
    public data: ActionData
  ) {}
}
