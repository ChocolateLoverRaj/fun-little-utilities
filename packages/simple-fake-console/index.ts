export * from './action'
export * from './console'
export * from './string'
