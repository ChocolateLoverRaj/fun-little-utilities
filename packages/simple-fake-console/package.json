{
  "name": "simple-fake-console",
  "version": "0.1.4",
  "description": "Create fake consoles for testing purpose",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://ksxnodeapps.github.io/fun-little-utilities/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodeapps/fun-little-utilities.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodeapps/fun-little-utilities/issues"
  },
  "keywords": [
    "console",
    "fake",
    "test"
  ],
  "dependencies": {
    "tslib": "^1.10.0",
    "@types/node": "^13.1.6"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
