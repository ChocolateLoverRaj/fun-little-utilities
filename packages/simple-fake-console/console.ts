import { Action, ActionType, ActionInstance } from './action'

export interface LogFunc {
  (...args: any[]): void
}

export interface Console {
  readonly log: LogFunc
  readonly info: LogFunc
  readonly error: LogFunc
  readonly warn: LogFunc
}

export interface ConsoleDatabase {
  getActions (): ReadonlyArray<Action>
}

export class ConsoleInstance implements Console, ConsoleDatabase {
  private readonly actions = Array<ActionInstance>()

  private readonly logfn = (type: ActionType): LogFunc => (...data) => {
    this.actions.push(new ActionInstance(type, data))
  }

  public readonly getActions = (): ReadonlyArray<Action> => this.actions

  public readonly log = this.logfn(ActionType.Log)
  public readonly info = this.logfn(ActionType.Info)
  public readonly error = this.logfn(ActionType.Error)
  public readonly warn = this.logfn(ActionType.Warn)
}
