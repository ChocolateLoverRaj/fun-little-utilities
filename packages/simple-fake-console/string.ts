import { ActionType } from './action'
import { ConsoleDatabase } from './console'

export function getString (options: getString.Options): string {
  const { console, types, wordSeparator = ' ', lineSeparator = '\n' } = options
  const typeSet = new Set(types)

  return console
    .getActions()
    .filter(x => typeSet.has(x.type))
    .map(x => x.data.join(wordSeparator))
    .join(lineSeparator)

}

export namespace getString {
  export interface Options {
    readonly console: ConsoleDatabase
    readonly types: Iterable<ActionType>
    readonly wordSeparator?: string
    readonly lineSeparator?: string
  }
}
