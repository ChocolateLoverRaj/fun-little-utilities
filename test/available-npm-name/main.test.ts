import { EventEmitter } from 'events'
import { resolve } from 'url'
import { ConsoleInstance, ActionType, getString } from 'simple-fake-console'
import delay from 'simple-delay'

import {
  Status,
  Process,
  Console,
  Fetch,
  NPM_REGISTRY,
  main
} from 'available-npm-name'

class FakeChunk implements Process.Chunk {
  constructor (
    private readonly text: string
  ) {}
  public toString () {
    return this.text
  }
}

class FakeStream extends EventEmitter implements Process.Stream {
  protected async * prvIterate () {
    yield * [
      '  abc', '\n',
      'de', 'f \n',
      'ghi\n',
      '  jkl\n',
      '@abc', '/def', '\n',
      '', '  \n',
      'mno\n',
      '    \n  ',
      'foo\n',
      'b', 'a', 'r'
    ].map(async (text, index) => {
      await delay(index * 10)
      return new FakeChunk(text)
    })
  }

  public async emitAllData () {
    for await (const chunk of this.prvIterate()) {
      this.emit('data', chunk)
    }
    this.emit('close')
  }
}

class FakeConsole implements Console.Mod {
  public readonly core = new ConsoleInstance()
  public readonly info = jest.fn(this.core.info)

  public getString () {
    return '\n' + getString({
      console: this.core,
      types: [ActionType.Info]
    }) + '\n'
  }
}

const availableNames = [
  'abc',
  'def',
  '@abc/def'
]

const occupiedNames = [
  'ghi',
  'jkl'
]

const erroredNames = [
  'mno',
  'foo',
  'bar'
]

const invalidNames = [
  '!!foo',
  'bar baz'
]

const hasUrl = (names: readonly string[], url: string) => names
  .map(name => resolve(NPM_REGISTRY, encodeURIComponent(name)))
  .includes(url)

const fetchImpl: Fetch.Fn = async url => {
  if (!url.startsWith(NPM_REGISTRY)) {
    throw new Error(`URL are supposed to start with ${NPM_REGISTRY}: ${url}`)
  }

  if (hasUrl(availableNames, url)) return { status: 404 }
  if (hasUrl(occupiedNames, url)) return { status: 200 }
  if (hasUrl(erroredNames, url)) return { status: -1 }

  throw new RangeError(`URL ${url} is unaccounted for`)
}

async function setup (args: readonly string[]) {
  const console = new FakeConsole()
  const fetch = jest.fn(fetchImpl)
  const stdin = new FakeStream()
  void stdin.emitAllData()
  const result = await main({
    argv: {
      _: args,
      registry: NPM_REGISTRY
    },
    process: {
      stdin
    },
    console,
    fetch
  })
  return { console, fetch, stdin, result }
}

describe('names are supplied via cli arguments', () => {
  describe('all names are available', () => {
    const args = [...availableNames]

    it('messages', async () => {
      const { console } = await setup(args)
      expect(console.getString()).toMatchSnapshot()
    })

    it('returns Status.Available', async () => {
      const { result } = await setup(args)
      expect(result).toBe(Status.Available)
    })

    it('calls fetch', async () => {
      const { fetch } = await setup(args)
      expect(fetch.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })
  })

  describe('all names are occupied', () => {
    const args = [...occupiedNames]

    it('messages', async () => {
      const { console } = await setup(args)
      expect(console.getString()).toMatchSnapshot()
    })

    it('returns Status.Occupied', async () => {
      const { result } = await setup(args)
      expect(result).toBe(Status.Occupied)
    })

    it('calls fetch', async () => {
      const { fetch } = await setup(args)
      expect(fetch.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })
  })

  describe('all names are causing errors', () => {
    const args = [...erroredNames]

    it('messages', async () => {
      const { console } = await setup(args)
      expect(console.getString()).toMatchSnapshot()
    })

    it('returns Status.NetworkError', async () => {
      const { result } = await setup(args)
      expect(result).toBe(Status.NetworkError)
    })

    it('calls fetch', async () => {
      const { fetch } = await setup(args)
      expect(fetch.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })
  })

  describe('all names are invalid', () => {
    const args = [...invalidNames]

    it('messages', async () => {
      const { console } = await setup(args)
      expect(console.getString()).toMatchSnapshot()
    })

    it('returns Status.InvalidName', async () => {
      const { result } = await setup(args)
      expect(result).toBe(Status.InvalidName)
    })

    it('calls fetch', async () => {
      const { fetch } = await setup(args)
      expect(fetch.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })
  })

  describe('all names either available or invalid', () => {
    const args = [...availableNames, ...invalidNames]

    it('messages', async () => {
      const { console } = await setup(args)
      expect(console.getString()).toMatchSnapshot()
    })

    it('returns Status.Available | Status.InvalidName', async () => {
      const { result } = await setup(args)
      expect(result).toBe(Status.Available | Status.InvalidName)
    })

    it('calls fetch', async () => {
      const { fetch } = await setup(args)
      expect(fetch.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })
  })

  describe('all names either occupied or causing errors', () => {
    const args = [...occupiedNames, ...erroredNames]

    it('messages', async () => {
      const { console } = await setup(args)
      expect(console.getString()).toMatchSnapshot()
    })

    it('returns Status.Occupied | Status.NetworkError', async () => {
      const { result } = await setup(args)
      expect(result).toBe(Status.Occupied | Status.NetworkError)
    })

    it('calls fetch', async () => {
      const { fetch } = await setup(args)
      expect(fetch.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })
  })

  describe('all status combined', () => {
    const args = [...availableNames, ...occupiedNames, ...erroredNames, ...invalidNames]

    it('messages', async () => {
      const { console } = await setup(args)
      expect(console.getString()).toMatchSnapshot()
    })

    it('returns Status.Available | Status.Occupied | Status.NetworkError | Status.InvalidName', async () => {
      const { result } = await setup(args)
      expect(result).toBe(Status.Available | Status.Occupied | Status.NetworkError | Status.InvalidName)
    })

    it('calls fetch', async () => {
      const { fetch } = await setup(args)
      expect(fetch.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })
  })
})

describe('names are supplied via stdin', () => {
  const args = [] as const

  describe('combine all status except InvalidName', () => {
    it('messages', async () => {
      const { console } = await setup(args)
      expect(console.getString()).toMatchSnapshot()
    })

    it('returns Status.Available | Status.Occupied | Status.NetworkError', async () => {
      const { result } = await setup(args)
      expect(result).toBe(Status.Available | Status.Occupied | Status.NetworkError)
    })

    it('calls fetch', async () => {
      const { fetch } = await setup(args)
      expect(fetch.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })

    it('calls console.info', async () => {
      const { console } = await setup(args)
      expect(console.info.mock.calls).toMatchSnapshot()
    })
  })
})
