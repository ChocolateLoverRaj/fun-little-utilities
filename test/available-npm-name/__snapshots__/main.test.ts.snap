// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch causes some errors calls console.info 1`] = `
Array [
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
]
`;

exports[`fetch causes some errors calls console.info 2`] = `
Array [
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
]
`;

exports[`fetch causes some errors calls fetch 1`] = `
Array [
  Array [
    "https://registry.npmjs.org/abc",
  ],
  Array [
    "https://registry.npmjs.org/def",
  ],
  Array [
    "https://registry.npmjs.org/ERROR1",
  ],
  Array [
    "https://registry.npmjs.org/ghi",
  ],
  Array [
    "https://registry.npmjs.org/ERROR2",
  ],
  Array [
    "https://registry.npmjs.org/jkl",
  ],
]
`;

exports[`fetch causes some errors messages 1`] = `
"
✅ abc is available
✅ def is available
"
`;

exports[`names are supplied via cli arguments all names are available calls console.info 1`] = `
Array [
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
  Array [
    "✅ @abc/def is available",
  ],
]
`;

exports[`names are supplied via cli arguments all names are available calls console.info 2`] = `
Array [
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
  Array [
    "✅ @abc/def is available",
  ],
]
`;

exports[`names are supplied via cli arguments all names are available calls fetch 1`] = `
Array [
  Array [
    "https://registry.npmjs.org/abc",
  ],
  Array [
    "https://registry.npmjs.org/def",
  ],
  Array [
    "https://registry.npmjs.org/%40abc%2Fdef",
  ],
]
`;

exports[`names are supplied via cli arguments all names are available messages 1`] = `
"
✅ abc is available
✅ def is available
✅ @abc/def is available
"
`;

exports[`names are supplied via cli arguments all names are causing errors calls console.info 1`] = `
Array [
  Array [
    "❓ mno ran into a network error",
  ],
  Array [
    "❓ foo ran into a network error",
  ],
  Array [
    "❓ bar ran into a network error",
  ],
]
`;

exports[`names are supplied via cli arguments all names are causing errors calls console.info 2`] = `
Array [
  Array [
    "❓ mno ran into a network error",
  ],
  Array [
    "❓ foo ran into a network error",
  ],
  Array [
    "❓ bar ran into a network error",
  ],
]
`;

exports[`names are supplied via cli arguments all names are causing errors calls fetch 1`] = `
Array [
  Array [
    "https://registry.npmjs.org/mno",
  ],
  Array [
    "https://registry.npmjs.org/foo",
  ],
  Array [
    "https://registry.npmjs.org/bar",
  ],
]
`;

exports[`names are supplied via cli arguments all names are causing errors messages 1`] = `
"
❓ mno ran into a network error
❓ foo ran into a network error
❓ bar ran into a network error
"
`;

exports[`names are supplied via cli arguments all names are invalid calls console.info 1`] = `
Array [
  Array [
    "👎 !!foo is an invalid name",
  ],
  Array [
    "👎 bar baz is an invalid name",
  ],
]
`;

exports[`names are supplied via cli arguments all names are invalid calls console.info 2`] = `
Array [
  Array [
    "👎 !!foo is an invalid name",
  ],
  Array [
    "👎 bar baz is an invalid name",
  ],
]
`;

exports[`names are supplied via cli arguments all names are invalid calls fetch 1`] = `Array []`;

exports[`names are supplied via cli arguments all names are invalid messages 1`] = `
"
👎 !!foo is an invalid name
👎 bar baz is an invalid name
"
`;

exports[`names are supplied via cli arguments all names are occupied calls console.info 1`] = `
Array [
  Array [
    "❌ ghi is occupied",
  ],
  Array [
    "❌ jkl is occupied",
  ],
]
`;

exports[`names are supplied via cli arguments all names are occupied calls console.info 2`] = `
Array [
  Array [
    "❌ ghi is occupied",
  ],
  Array [
    "❌ jkl is occupied",
  ],
]
`;

exports[`names are supplied via cli arguments all names are occupied calls fetch 1`] = `
Array [
  Array [
    "https://registry.npmjs.org/ghi",
  ],
  Array [
    "https://registry.npmjs.org/jkl",
  ],
]
`;

exports[`names are supplied via cli arguments all names are occupied messages 1`] = `
"
❌ ghi is occupied
❌ jkl is occupied
"
`;

exports[`names are supplied via cli arguments all names either available or invalid calls console.info 1`] = `
Array [
  Array [
    "👎 !!foo is an invalid name",
  ],
  Array [
    "👎 bar baz is an invalid name",
  ],
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
  Array [
    "✅ @abc/def is available",
  ],
]
`;

exports[`names are supplied via cli arguments all names either available or invalid calls console.info 2`] = `
Array [
  Array [
    "👎 !!foo is an invalid name",
  ],
  Array [
    "👎 bar baz is an invalid name",
  ],
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
  Array [
    "✅ @abc/def is available",
  ],
]
`;

exports[`names are supplied via cli arguments all names either available or invalid calls fetch 1`] = `
Array [
  Array [
    "https://registry.npmjs.org/abc",
  ],
  Array [
    "https://registry.npmjs.org/def",
  ],
  Array [
    "https://registry.npmjs.org/%40abc%2Fdef",
  ],
]
`;

exports[`names are supplied via cli arguments all names either available or invalid messages 1`] = `
"
👎 !!foo is an invalid name
👎 bar baz is an invalid name
✅ abc is available
✅ def is available
✅ @abc/def is available
"
`;

exports[`names are supplied via cli arguments all names either occupied or causing errors calls console.info 1`] = `
Array [
  Array [
    "❌ ghi is occupied",
  ],
  Array [
    "❌ jkl is occupied",
  ],
  Array [
    "❓ mno ran into a network error",
  ],
  Array [
    "❓ foo ran into a network error",
  ],
  Array [
    "❓ bar ran into a network error",
  ],
]
`;

exports[`names are supplied via cli arguments all names either occupied or causing errors calls console.info 2`] = `
Array [
  Array [
    "❌ ghi is occupied",
  ],
  Array [
    "❌ jkl is occupied",
  ],
  Array [
    "❓ mno ran into a network error",
  ],
  Array [
    "❓ foo ran into a network error",
  ],
  Array [
    "❓ bar ran into a network error",
  ],
]
`;

exports[`names are supplied via cli arguments all names either occupied or causing errors calls fetch 1`] = `
Array [
  Array [
    "https://registry.npmjs.org/ghi",
  ],
  Array [
    "https://registry.npmjs.org/jkl",
  ],
  Array [
    "https://registry.npmjs.org/mno",
  ],
  Array [
    "https://registry.npmjs.org/foo",
  ],
  Array [
    "https://registry.npmjs.org/bar",
  ],
]
`;

exports[`names are supplied via cli arguments all names either occupied or causing errors messages 1`] = `
"
❌ ghi is occupied
❌ jkl is occupied
❓ mno ran into a network error
❓ foo ran into a network error
❓ bar ran into a network error
"
`;

exports[`names are supplied via cli arguments all status combined calls console.info 1`] = `
Array [
  Array [
    "👎 !!foo is an invalid name",
  ],
  Array [
    "👎 bar baz is an invalid name",
  ],
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
  Array [
    "✅ @abc/def is available",
  ],
  Array [
    "❌ ghi is occupied",
  ],
  Array [
    "❌ jkl is occupied",
  ],
  Array [
    "❓ mno ran into a network error",
  ],
  Array [
    "❓ foo ran into a network error",
  ],
  Array [
    "❓ bar ran into a network error",
  ],
]
`;

exports[`names are supplied via cli arguments all status combined calls console.info 2`] = `
Array [
  Array [
    "👎 !!foo is an invalid name",
  ],
  Array [
    "👎 bar baz is an invalid name",
  ],
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
  Array [
    "✅ @abc/def is available",
  ],
  Array [
    "❌ ghi is occupied",
  ],
  Array [
    "❌ jkl is occupied",
  ],
  Array [
    "❓ mno ran into a network error",
  ],
  Array [
    "❓ foo ran into a network error",
  ],
  Array [
    "❓ bar ran into a network error",
  ],
]
`;

exports[`names are supplied via cli arguments all status combined calls fetch 1`] = `
Array [
  Array [
    "https://registry.npmjs.org/abc",
  ],
  Array [
    "https://registry.npmjs.org/def",
  ],
  Array [
    "https://registry.npmjs.org/%40abc%2Fdef",
  ],
  Array [
    "https://registry.npmjs.org/ghi",
  ],
  Array [
    "https://registry.npmjs.org/jkl",
  ],
  Array [
    "https://registry.npmjs.org/mno",
  ],
  Array [
    "https://registry.npmjs.org/foo",
  ],
  Array [
    "https://registry.npmjs.org/bar",
  ],
]
`;

exports[`names are supplied via cli arguments all status combined messages 1`] = `
"
👎 !!foo is an invalid name
👎 bar baz is an invalid name
✅ abc is available
✅ def is available
✅ @abc/def is available
❌ ghi is occupied
❌ jkl is occupied
❓ mno ran into a network error
❓ foo ran into a network error
❓ bar ran into a network error
"
`;

exports[`names are supplied via stdin combine all status except InvalidName calls console.info 1`] = `
Array [
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
  Array [
    "❌ ghi is occupied",
  ],
  Array [
    "❌ jkl is occupied",
  ],
  Array [
    "✅ @abc/def is available",
  ],
  Array [
    "❓ mno ran into a network error",
  ],
  Array [
    "❓ foo ran into a network error",
  ],
  Array [
    "❓ bar ran into a network error",
  ],
]
`;

exports[`names are supplied via stdin combine all status except InvalidName calls console.info 2`] = `
Array [
  Array [
    "✅ abc is available",
  ],
  Array [
    "✅ def is available",
  ],
  Array [
    "❌ ghi is occupied",
  ],
  Array [
    "❌ jkl is occupied",
  ],
  Array [
    "✅ @abc/def is available",
  ],
  Array [
    "❓ mno ran into a network error",
  ],
  Array [
    "❓ foo ran into a network error",
  ],
  Array [
    "❓ bar ran into a network error",
  ],
]
`;

exports[`names are supplied via stdin combine all status except InvalidName calls fetch 1`] = `
Array [
  Array [
    "https://registry.npmjs.org/abc",
  ],
  Array [
    "https://registry.npmjs.org/def",
  ],
  Array [
    "https://registry.npmjs.org/ghi",
  ],
  Array [
    "https://registry.npmjs.org/jkl",
  ],
  Array [
    "https://registry.npmjs.org/%40abc%2Fdef",
  ],
  Array [
    "https://registry.npmjs.org/mno",
  ],
  Array [
    "https://registry.npmjs.org/foo",
  ],
  Array [
    "https://registry.npmjs.org/bar",
  ],
]
`;

exports[`names are supplied via stdin combine all status except InvalidName messages 1`] = `
"
✅ abc is available
✅ def is available
❌ ghi is occupied
❌ jkl is occupied
✅ @abc/def is available
❓ mno ran into a network error
❓ foo ran into a network error
❓ bar ran into a network error
"
`;
